### PR TITLE
feat: Add config option to customize logo and search icon

### PR DESCRIPTION
**Pull request by KattMingMing**
Adds a lightweight way to customize the homepage logo, search icon, and favicon. Supports light and dark theme icons.

Future additions:
~Support light and dark theme for both icons (if this is something we want now I can update PR)~
- Allow custom CSS to override theme

### CODE REVIEW

File: schema/site.schema.json [hunk 1]
```diff
@@ -141,6 +141,23 @@
...
     },
+    "branding": {
+      "description": "Customize Sourcegraph homepage logo and search icon.",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "logo": {
+          "description": "The image URL to be used on the search homepage.",
+          "type": "string",
+          "default": ""
```
Comment (felixfbecker) [inline schema/site.schema.json]: In JSON/JS/TS, empty strings should generally not be allowed.

File: cmd/frontend/internal/app/jscontext/jscontext.go [hunk 2]
```diff
@@ -76,6 +76,8 @@ type JSContext struct {
 	ResetPasswordEnabled bool `json:"resetPasswordEnabled"`
 
 	AuthProviders []authProviderInfo `json:"authProviders"`
+
+	Branding schema.Branding `json:"branding"`
```
Comment (felixfbecker) [inline cmd/frontend/internal/app/jscontext/jscontext.go]: In the TS definitions, this is typed as possibly undefined, but here it seems it is always defined (but with potentially empty strings). This should probably be a pointer?

File: web/src/globals.d.ts [hunk 3]
```diff
@@ -109,6 +109,14 @@ interface SourcegraphContext {
         isBuiltin: boolean
         authenticationURL?: string
     }[]
+
+    /** Custom branding for the homepage and search icon. */
+    branding?: {
+        /** The URL to the logo used on the search homepage. */
+        logo: string
+        /** The URL to the icon used as the search logo */
+        favicon: string
```
Comment (felixfbecker) [inline web/src/globals.d.ts]: Is it possible to only customize the logo, but not the favicon? The JSON schema currently allows that, but not the TS types

File: schema/site.schema.json [hunk 4]
```diff
@@ -141,6 +141,23 @@
...
       "group": "Security"
     },
+    "branding": {
+      "description": "Customize Sourcegraph homepage logo and search icon.",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "logo": {
+          "description": "The image URL to be used on the search homepage.",
+          "type": "string",
```
Comment (felixfbecker) [inline schema/site.schema.json]: Suggestion: add `"format": "uri"`

File: schema/site.schema.json [hunk 5]
```diff
@@ -141,6 +141,23 @@
...
+          "default": ""
+        },
+        "favicon": {
+          "description": "The image URL to be used as the search icon.",
```
Comment (felixfbecker) [inline schema/site.schema.json]: `favicon` seems like the wrong name for this - a favicon is the icon used by the browser in the tab and bookmarks

File: CHANGELOG.md [hunk 6]
```diff
@@ -23,6 +23,7 @@ All notable changes to Sourcegraph are documented in this file.
   To enable, open the user menu in the top right and make sure the theme dropdown is set to "System".
   This is currently supported on macOS Mojave with Safari Technology Preview 68 and later.
 - The `github.exclude` setting was added to the [GitHub external service config](https://docs.sourcegraph.com/admin/external_service/github#configuration) to allow excluding repositories yielded by `github.repos` or `github.repositoryQuery` from being synced.
+- Enterprise admins can now customize the appearance of the homepage and search icon.
```
Comment (sqs) [inline CHANGELOG.md]: Is this under 3.2? Should be under 3.2.1 right?

File: schema/site.schema.json (hunk 4 above)
Comment (KattMingMing) [reply schema/site.schema.json]: `uri` isn't a supported type, is `string` okay?

File: schema/site.schema.json [hunk 7]
```diff
@@ -141,6 +141,45 @@
...
+          "type": "object",
+          "properties": {
+            "logo": {
+              "description": "The URL to the image used on the homepage. This will replace the Sourcegraph logo on the homepage. Maximum width: 320px. We recommend using the following file formats: SVG, PNG",
+              "type": "string"
+            },
+            "symbol": {
+              "description": "The URL to the symbol used as the search icon. Recommended size: 24x24px. We recommend using the following file formats: SVG, PNG, ICO",
+              "type": "string"
+            }
```
Comment (felixfbecker) [inline schema/site.schema.json]: I would remove this duplication by defining this object at the top and referencing it with `$ref`. See https://sourcegraph.sgdev.org/search?q=%22%24ref%22+f:%5C.json%24#1

File: web/src/globals.d.ts [hunk 8]
```diff
@@ -109,6 +109,27 @@ interface SourcegraphContext {
...
+    branding?: {
+        /** The URL of the favicon to be used for your instance */
+        favicon?: string
+
+        /** Override style for light themes */
+        light?: {
+            /** The URL to the logo used on the homepage */
+            logo?: string
+            /** The URL to the symbol used as the search icon */
+            symbol?: string
```
Comment (felixfbecker) [inline web/src/globals.d.ts]: I would define an interface for this object to remove the duplication

File: schema/site.schema.json (hunk 4 above)
Comment (KattMingMing) [reply schema/site.schema.json]: Added 👍

**Pull request merged**